### PR TITLE
Move forms to RTK

### PR DESCRIPTION
**Pull request by Georgi2704**

### CODE REVIEW

**Comment by DutchBen** [comment]
Add a changeset

File: packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx [hunk 1]
```diff
@@ -13,13 +12,18 @@ import {
     EuiText,
 } from '@elastic/eui';
 
-import { PATH_TASKS, WfoError, WfoLoading } from '@/components';
-import { PATH_WORKFLOWS } from '@/components';
+import { PATH_TASKS, PATH_WORKFLOWS, WfoError, WfoLoading } from '@/components';
+import UserInputFormWizard from '@/components/WfoForms/UserInputFormWizard';
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx:16]: Make this a named import
```suggestion
import { UserInputFormWizard } from '@/components/WfoForms/UserInputFormWizard';
```

File: packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx [hunk 2]
```diff
@@ -133,13 +137,12 @@ export const WfoStartProcessPage = ({
                         throw e;
                     },
                 )
-                .catch((error: AxiosError) => {
-                    if (error?.response?.status !== 510) {
-                        if (error?.response?.status === 400) {
+                .catch((error) => {
+                    if (error?.status !== 510) {
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx:141]: Can you use the HttpStatus enum here aswell, and other places we check status codes? I do prefer to keep this checking consitent. So we choose on or the other

File: packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { BaseQueryTypes, orchestratorApi } from '@/rtk';
+
+const PROCESS_ENDPOINT = 'processes';
+const RESUME_ENDPOINT = 'resume';
+const FORMS_ENDPOINT = 'surf/forms/'; // It is still being used by example-wfo-uo
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts:5]:
```suggestion
const FORMS_ENDPOINT = 'surf/forms/'; // It is still being used by example-wfo-ui
```

File: packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { BaseQueryTypes, orchestratorApi } from '@/rtk';
+
+const PROCESS_ENDPOINT = 'processes';
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts:3]: Are there scenarios where these would be changed so we should add them to the configuration? I guess not right?

File: packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+const RESUME_ENDPOINT = 'resume';
+const FORMS_ENDPOINT = 'surf/forms/'; // It is still being used by example-wfo-uo
+
+const formsApi = orchestratorApi.injectEndpoints({
+    endpoints: (build) => ({
+        startProcess: build.mutation<
+            unknown,
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts:10]: This 'unknow' type is the return value right? Are you sure it's unknown? If there's not return value you can say void I think

File: packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            }),
+            extraOptions: {
+                baseQueryType: BaseQueryTypes.fetch,
+            },
+        }),
+        resumeProcess: build.mutation<
+            unknown,
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts:26]: Same comment as above

File: packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            }),
+            extraOptions: {
+                baseQueryType: BaseQueryTypes.fetch,
+            },
+        }),
+        startForm: build.mutation<
+            unknown,
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts:42]: And again

**Review by DutchBen — APPROVED** [review]

File: packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts (hunk 5 above)
Comment (Georgi2704) [reply packages/orchestrator-ui-components/src/rtk/endpoints/forms.ts:10]: `    startProcess = (
        workflowName: string,
        processInput: object,
    ): Promise<unknown> => {
        return this.postPutJson<unknown>(
            `${PROCESS_ENDPOINT}${workflowName}`,
            processInput,
            'post',
            false,
            true,
        );
    };`
    
    This was the axios version, that is why I used unknown

File: packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx [hunk 8]
```diff
@@ -110,17 +115,15 @@ export const WfoStartProcessPage = ({
...
-            const startProcessPromise = apiClient
-                .startProcess(
-                    processName,
-                    startProcessPayload
-                        ? [startProcessPayload, ...processInput]
-                        : [...processInput],
-                )
+            const startProcessPromise = startProcess({
+                workflowName: processName,
+                userInputs: startProcessPayload
```
Comment (tjeerddie) [inline packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx:120]: I don't like multiline conditional operator, move this to a const above the `const startProcessPromise`, might fix the multiline.

**Review by tjeerddie — APPROVED** [review]

File: packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx (hunk 8 above)
Comment (tjeerddie) [reply packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx:120]: nvm, forgetting its because our small 80 line length...

**Pull request merged**